src/Functions/formatRow.cpp

```diff
@@ -0,0 +1,145 @@
+#include <memory>
+#include <Columns/ColumnString.h>
+#include <DataStreams/materializeBlock.h>
+#include <DataTypes/DataTypeString.h>
+#include <Formats/FormatFactory.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunctionImpl.h>
+#include <IO/WriteBufferFromVector.h>
+#include <IO/WriteHelpers.h>
+#include <Processors/Formats/IOutputFormat.h>
+#include <ext/map.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int NOT_IMPLEMENTED;
+    extern const int UNKNOWN_FORMAT;
+}
+
+
+/** formatRow(<format>, x, y, ...) is a function that allows you to use RowOutputFormat over
+  * several columns to generate a string per row, such as CSV, TSV, JSONEachRow, etc.
+  * formatRowNoNewline(...) trims the newline character of each row.
+  */
+
+template <bool no_newline>
+class FunctionFormatRow : public IFunction
+{
+public:
+    static constexpr auto name = no_newline ? "formatRowNoNewline" : "formatRow";
+
+    FunctionFormatRow(const String & format_name_, const Context & context_) : format_name(format_name_), context(context_)
+    {
+        if (!FormatFactory::instance().getAllFormats().count(format_name))
+            throw Exception("Unknown format " + format_name, ErrorCodes::UNKNOWN_FORMAT);
+
+        static constexpr auto supported_formats = {"CSV", "TSV", "JSONEachRow", "TabSeparated", "TabSeparatedRaw", "TSKV"};
```

Comment:
We can simply support all the formats, why we need these limitations?

For "no newline" mode we can check that the last character is newline and trim it. And it's ok even to have some implementation-specific behaviour for binary formats.

Reply:
Ok. I thought formats with prefix/suffix like `CSVWithNames` might be counter-intuitive to use...